fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java

```diff
@@ -610,48 +610,90 @@ public boolean validateData(Object srcValue, EtlJobConfig.EtlColumn etlColumn, C
         return dataframe;
     }
 
+    private StructType constructSrcSchema(EtlJobConfig.EtlFileGroup fileGroup, EtlJobConfig.EtlIndex baseIndex) {
+        List<String> dataSrcColumns = fileGroup.fileFieldNames;
+        if (dataSrcColumns == null) {
+            // if there is no source columns info
+            // use base index columns as source columns
+            dataSrcColumns = new ArrayList<>();
+            for (EtlJobConfig.EtlColumn column : baseIndex.columns) {
+                dataSrcColumns.add(column.columnName);
+            }
+        }
+        List<String> srcColumnsWithColumnsFromPath = new ArrayList<>();
+        srcColumnsWithColumnsFromPath.addAll(dataSrcColumns);
+        if (fileGroup.columnsFromPath != null) {
+            srcColumnsWithColumnsFromPath.addAll(fileGroup.columnsFromPath);
+        }
+        return createScrSchema(srcColumnsWithColumnsFromPath);
```

Comment:
Please fix the function name.
createScrSchema -> createSrcSchema

Reply:
fixed, PTAL